src/cloudant/database.py

```diff
@@ -1098,3 +1101,124 @@ def get_query_result(self, selector, fields=None, raw_result=False,
+            specified count. In case of a grouped search, this parameter limits
+            the number of documents per group.
+        :param ranges: A JSON facet syntax that reuses the standard Lucene
+            syntax to return counts of results which fit into each specified
+            category. Inclusive range queries are denoted by brackets.
+            Exclusive range queries are denoted by curly brackets.
+            For example ``ranges={"price":{"cheap":"[0 TO 100]"}}`` has an
+            inclusive range of 0 to 100.
+            Requires the index to have faceting enabled.
+        :param sort: A JSON string of the form ``fieldname<type>`` for ascending
+            or ``-fieldname<type>`` for descending sort order. Fieldname is the
+            name of a string or number field and type is either number or string
+            or a JSON array of such strings. The type part is optional and
+            defaults to number.
+        :param str stale: Allow the results from a stale index to be used. This
+            makes the request return immediately, even if the index has not been
+            completely built yet.
+        :param list highlight_fields: A list of fields which should be
+            highlighted.
+        :param str highlight_pre_tag: A string inserted before the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param str highlight_post_tag: A string inserted after the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param int highlight_number: A number of fragments returned in
+            highlights. If the search term occurs less often than the number of
+            fragments specified, longer fragments are returned.  Default is 1.
+        :param int highlight_size: A number of characters in each fragment for
+            highlights.  Defaults to 100 characters.
+        :param list include_fields: A list of field names to include in search
+            results. Any fields included must have been indexed with the
+            ``store:true`` option.
+
+        :returns: Search query result data in JSON format
+        """
+        data = dict(self)
+        data.update(query=query)
+        data.update(kwargs)
+
+        # Validate query arguments and values
+        for key, val in iteritems_(data):
+            if key not in list(SEARCH_INDEX_ARGS.keys()):
+                msg = 'Invalid argument: {0}'.format(key)
+                raise CloudantArgumentError(msg)
+            if not isinstance(val, SEARCH_INDEX_ARGS[key]):
+                msg = (
+                    'Argument {0} is not an instance of expected type: {1}'
+                ).format(key, SEARCH_INDEX_ARGS[key])
+                raise CloudantArgumentError(msg)
+        # Execute query search
+        headers = {'Content-Type': 'application/json'}
+        ddoc = DesignDocument(self, ddoc_id)
+        resp = ddoc.r_session.post(
```

Comment:
Since you're in the CloudantDatabase, you should use the CloudantDatabase `r_session`.  The example I provided before should have used `self.r_session` instead of `ddoc.r_session`.